pallets/order-book/src/order_book.rs

```diff
@@ -1167,4 +1168,59 @@ impl<T: crate::Config + Sized> OrderBook<T> {
 
         (market_amount, amount)
     }
+
+    pub fn market_depth(
+        &self,
+        side: PriceVariant,
+        volume_limit: Option<OrderAmount>,
+        data: &mut impl DataLayer<T>,
+    ) -> Vec<(OrderPrice, OrderVolume)> {
+        match side {
+            PriceVariant::Buy => Self::get_market_depth(
+                volume_limit,
+                data.get_aggregated_bids(&self.order_book_id).iter().rev(),
+            ),
+            PriceVariant::Sell => Self::get_market_depth(
+                volume_limit,
+                data.get_aggregated_asks(&self.order_book_id).iter(),
+            ),
+        }
+    }
+
+    fn get_market_depth<'a>(
```

Comment:
minor thing, but imho something like `stripped_market_data` or `limited_market_data` or `bounded_market_data` would be more precise for the implementation